motionplan/constraint.go

```diff
@@ -131,67 +131,112 @@ func (c *constraintHandler) CheckConstraints(cInput *ConstraintInput) (bool, flo
 // Collisions specified as collisionSpecifications will also be ignored
 // if reportDistances is false, this check will be done as fast as possible, if true maximum information will be available for debugging.
 func newSelfCollisionConstraint(
-	frame referenceframe.Frame,
+	frame *solverFrame,
 	observationInput map[string][]referenceframe.Input,
 	collisionSpecifications []*Collision,
 	reportDistances bool,
 ) (Constraint, error) {
-	return newCollisionConstraint(frame, nil, observationInput, collisionSpecifications, reportDistances)
+	frameInputs, err := frame.mapToSlice(observationInput)
+	if err != nil {
+		return nil, err
+	}
+
+	// create robot collision entities
+	movingGeometries, err := frame.Geometries(frameInputs)
```

Comment:
it feels pretty bad to be computing the moving geometries three times, once in each function.  The thing we could do to get rid of this problem is having all this logic live in the `plannerOptions` func creating them, and interact directly with `newCollisionConstraint`.  Thoughts?

Reply:
Such as up in `createCollisionConstraint` in `plannerOptions.go`? I think that's a good idea. Feels like it improves the generality of these functions a bit.

Which leads me to another thought: If the degree of difference between `newObstacleConstraint` and `newSelfCollisionConstraint` and `newRobotConstraint` is that we're just changing the second argument to `newCollisionConstraint` for each function...why not simplify the process to get the required geometries and cut down on the line count there as well?

Reply:
Yeah I think I'm going to go through with this change.  I'll put it up and see what you think, should end up being a lot more concise